src/functional/ffront/type_info.py

```diff
@@ -131,7 +157,7 @@ def _as_field(dtype: ts.ScalarType, path: tuple[int, ...]) -> ts.FieldType:
         returns=ts.DeferredType(constraint=None),
     )
 
-    yield from function_signature_incompatibilities(function_type, args, kwargs)
+    yield from function_signature_incompatibilities(function_type, new_args, kwargs)
```

Comment:
```suggestion
    yield from function_signature_incompatibilities(function_type, promote_zero_dims(args, function_type), kwargs)
```
With my current understanding this should make the changes above superfluous.